From b0eaa2fabde7cb8807dbd71f2f00efcb3266958c Mon Sep 17 00:00:00 2001
From: Guoliang Ji <guoliang.ji@intel.com>
Date: Thu, 2 May 2013 16:43:52 +0800
Subject: Fix the fps less than 30 issue

BZ: 50971

refine the call sequence of the fillOutputBuffers and drainInputBuffers to
enhance the fps performance

Category: aosp improvement
Domain: Video.Media-video recording
Origin: internal
Upstream-Candidate: yes

Change-Id: I60bef371cba0cae3592c6512a3df0070cc909186
Orig-Change-Id: I1726159e6b6133d29519e28578f25b8a09cfc6e5
Signed-off-by: jiguoliang <guoliang.ji@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
Signed-off-by: Yuanjun Huang <yuanjun.huang@intel.com>
---
 media/libstagefright/OMXCodec.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/media/libstagefright/OMXCodec.cpp b/media/libstagefright/OMXCodec.cpp
index dfc9197..e58a417 100644
--- a/media/libstagefright/OMXCodec.cpp
+++ b/media/libstagefright/OMXCodec.cpp
@@ -3884,13 +3884,13 @@ status_t OMXCodec::read(
             mPaused = false;
         }
 
-        drainInputBuffers();
 
         if (mState == EXECUTING) {
             // Otherwise mState == RECONFIGURING and this code will trigger
             // after the output port is reenabled.
             fillOutputBuffers();
         }
+        drainInputBuffers();
     }
 
     if (seeking) {
-- 
1.7.4.1

